packages/web/components/pool-detail/concentrated.tsx

```diff
@@ -86,19 +86,81 @@ export const ConcentratedLiquidityPool: FunctionComponent<{ poolId: string }> =
       ? pool.concentratedLiquidityPoolInfo.currentPrice
       : undefined;
 
-    const userPositions = osmosisQueries.queryAccountsPositions.get(
-      account?.address ?? ""
-    ).positions;
+    const userPositionsInPool = osmosisQueries.queryAccountsPositions
+      .get(account?.address ?? "")
+      .positionsInPool(poolId);
 
-    const userPositionsInPool = userPositions.filter(
-      (position) => position.poolId === poolId
-    );
     const userHasPositionInPool = userPositionsInPool.length > 0;
 
-    const rewardedPositions = userPositions.filter(
+    const rewardedPositions = userPositionsInPool.filter(
       (position) => position.hasClaimableRewards
     );
 
+    const onClickCollectAllRewards = () => {
```

Comment:
Is all of this function unique to CL? Could any of it be reused in staking? If so, can we extract into a function that is reusable in staking?

Reply:
It's unique to CL- use of the word "rewards" here refers to the spread and incentive rewards a user's position accrues for providing liquidity in a CL pool. Vs inflationary rewards received for locking OSMO as staked OSMO.

Let me know if you have any other questions 👍